src/testE2E/codecatalyst/client.test.ts

```diff
@@ -279,6 +279,25 @@ describe('Test how this codebase uses the Code Catalyst API', function () {
         }
     })
 
+    describe('Repositories', function () {
+        it('can filter out non-internal repositories (eg: github)', async function () {
```

Comment:
The filter logic is the non-trivial part, right? So this is an example of something that doesn't need end-to-end coverage. We just need to extract the filter logic out of `listSourceRepositories` and test the logic. Testing the call to `listSourceRepositories` is less interesting.